src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Input/Commands/AbstractGenerateNuGetPackageCommand.cs

```diff
@@ -23,18 +40,66 @@ protected AbstractGenerateNuGetPackageCommand(UnconfiguredProject unconfiguredPr
 
         protected override Task<CommandStatusResult> GetCommandStatusAsync(IProjectTree node, bool focused, string commandText, CommandStatus progressiveStatus) =>
             ShouldHandle(node) ?
-                GetCommandStatusResult.Handled(GetCommandText(), CommandStatus.Enabled) :
+                GetCommandStatusResult.Handled(GetCommandText(), IsReadyToBuild() ? CommandStatus.Enabled : CommandStatus.Supported) :
                 GetCommandStatusResult.Unhandled;
 
+        private bool IsReadyToBuild()
+        {
+            return ErrorHandler.Succeeded(_buildManager.QueryBuildManagerBusy(out int busy)) && busy == 0;
```

Comment:
The out parameter for this API is a magic number, not an HRESULT. I checked all API usages on index, and we compare it with 0 at all places.

I'll switch to ErrorHandler.ThrowOnFailure to avoid catching exceptions.